server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -1,67 +1,75 @@
+import os
 import logging
 import pytest
 
-from pbench import get_pbench_logger
+from pathlib import Path
+
 from pbench.process_tb import ProcessTb, Results
-from pbench.test import tarball, bad_tarball, receive_dir
+from pbench.test import tarball, bad_tarball, receive_dir, fixture_path
 
 
 class MockConfig:
-    def __init__(self):
-        self.TS = "run-1970-01-02T00:00:00.000000"
+    TS = "run-1970-01-02T00:00:00.000000"
 
     def get(self, section: str, option: str):
         pass
 
 
 class TestProcessTb:
     @pytest.fixture(autouse=True)
-    def config_and_logger(self, valid_config):
+    def logger(self, test_logger):
         # Setup the configuration and logger
-        global tmp
-        config, tmp = valid_config
-        self.logger = get_pbench_logger("pbench", config)
+        self.logger = test_logger
```

Comment:
Rather than invoking the `test_logger` fixture inside another fixture and caching the result in the class, wouldn't it be better to just reference fixture directly where it's needed (instead of using the class attribute)?

Reply:
It is used in all tests so I thought of caching it.

Reply:
PyTest caches the _value_ of the fixture automatically -- that is, if you reference the fixture symbol more than once, the fixture function is not called a second time -- so there's no advantage to caching the result yourself.  The question is, do you like accessing it via `self.logger` better than via `test_logger`?...the latter requires listing the fixture in the function signature, while the former requires an additional fixture which adds the attribute to the class.  I prefer the latter....

Reply:
Both of them sounds good to me :)